aws-android-sdk-mobile-client/src/main/java/com/amazonaws/mobile/client/AWSMobileClient.java

```diff
@@ -483,11 +484,14 @@ public void run() {
                     mContext = context.getApplicationContext();
                     mStore = new AWSMobileClientStore(AWSMobileClient.this);
 
+                    AWSMobileClient.this.awsConfiguration = awsConfiguration;
```

Comment:
This is intentionally done last in `_initialize()` to validate that the provided configuration is correct. It seems that you are doing this ahead of time so that `registerConfigSignInProviders()` can take effect. I think the better approach would be to let `registerConfigSignInProviders()` method take in a configuration object as a parameter and pass it in.

Reply:
@raphkim I had been using my local branch version and recently switched back to using the latest implementation via gradle dependency on 2.23.0. The problem returned. Without setting AWSMobileClient.this.awsConfiguration = awsConfiguration at the line shown, earlier than is done in the latest, there are use cases where this.awsConfiguration is null, causing the UX to fail to initialize properly and register sign in providers. I will update my local copy and test.

Reply:
By design, `AWSMobileClient#awsConfiguration` should be `null` if `AWSMobileClient#initialize()` did not trigger a success callback yet. This is to prevent any accidental usage of an invalid configuration that previously failed to initialize mobile client in other parts of the code. If there is any need for configuration object during the initialization process, then it should use the configuration instance that was passed in as a parameter rather than the instance variable of mobile client instance.